processing/src/main/java/org/apache/druid/segment/join/JoinConditionAnalysis.java

```diff
@@ -79,7 +79,7 @@ private JoinConditionAnalysis(
                                                                 .allMatch(expr -> expr.isLiteral() && expr.eval(
                                                                     ExprUtils.nilBindings()).asBoolean());
     canHashJoin = nonEquiConditions.stream().allMatch(Expr::isLiteral);
-    rightKeyColumns = getEquiConditions().stream().map(Equality::getRightColumn).distinct().collect(Collectors.toSet());
+    rightKeyColumns = getEquiConditions().stream().map(Equality::getRightColumn).collect(Collectors.toSet());
```

Comment:
What's the impact of removing the `distinct()` call (is it just unnecessary since they're being collected to a set already?)

Reply:
should be a no-op. intelliJ suggested it's un-necessary since we're collecting to a set